components/canvas/ProcessHierarchy/useProcessHierarchyTabs.tsx

```diff
@@ -0,0 +1,65 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+
+export const useProcessHierarchyTabs = () => {
+  const { query, pathname, push } = useRouter();
+
+  const [activeTab, setActiveTab] = useState(Number(query.tab) || 0);
```

Comment:
Instead of having two sources of truth (`activeTab` and `query.tab`), I would stick to just reading `query.tab`:
```ts
const activeTab = Number(query.tab) || 0
```